src/network/hobbits/rpc/codec.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * @module network
+ */
+
+import {deserialize, serialize} from "@chainsafe/ssz";
+
+import {
+  Hello, Goodbye, RequestBody, WireRequest, GetStatus, GetBlockHeaders, BlockBodies, GetBlockBodies, BlockHeaders,
+} from "./messages";
+import {RequestId, Method} from "../constants";
+
+// Encode
+
+export function encodeRequest(id: RequestId, method: Method, body: RequestBody): Buffer {
```

Comment:
isn't this duplicate of one in libp2p?

Reply:
Not fully. `GetBlockHeaders`, `GetBlockHeaders` has extra `direction` field and there is no `BeaconStatesRequest` in `hobbits`.

Reply:
You could try to abstract it though. Direction can be optional param easily